lib/kernels/src/cuda/linear_kernels.cu

```diff
@@ -13,76 +13,78 @@
-}
-
 namespace Kernels {
 namespace Linear {
 
-bool use_activation(ActiMode mode) {
-  switch (mode) {
-    case AC_MODE_RELU:
-    case AC_MODE_SIGMOID:
-    case AC_MODE_TANH:
-      return true;
-    case AC_MODE_NONE:
-      return false;
-    default:
-      assert(0);
+// what's the float * one_ptr
+LinearPerDeviceState
+    init_kernel(PerDeviceFFHandle handle, Allocator allocator, float *one_ptr;
+                Activation activation,
+                Regularizer regularizer,
+                bool use_bias,
+                DataType input_type,
+                DataType weight_type,
+                DataType output_type,
+                int batch_size,
+                int channel) {
+  ffTensorDescriptor_t outputTensor;
+  ffActivationDescriptor_t actiDesc;
+  checkCUDNN(cudnnCreateTensorDescriptor(&outputTensor));
+  checkCUDNN(cudnnCreateActivationDescriptor(&actiDesc));
+  checkCUDNN(cudnnSetTensor4dDescriptor(outputTensor,
+                                        CUDNN_TENSOR_NCHW,
+                                        ff_to_cudnn_datatype(output_type),
+                                        batch_size,
+                                        channel,
+                                        1,
+                                        1));
+  cudnnActivationMode_t mode;
+  switch (activation) {
+    case RELU:
+      mode = CUDNN_ACTIVATION_RELU;
       break;
+    case SIGMOID:
+      mode = CUDNN_ACTIVATION_SIGMOID;
+      break;
+    case TANH:
+      mode = CUDNN_ACTIVATION_TANH;
+      break;
+    case GELU:
+      mode = CUDNN_ACTIVATION_GELU;
+      break;
+    default:
+      // Unsupported activation mode
+      assert(false);
   }
-  return false;
-}
+  checkCUDNN(
+      cudnnSetActivationDescriptor(actiDesc, mode, CUDNN_PROPAGATE_NAN, 0.0));
+  checkCUDNN(
+      cudnnSetTensorDescriptorFromArrayShape(outputTensor, output_shape));
 
-void init_kernel(LinearPerDeviceState *m, int batch_size, int channel) {
-  if (use_activation(m->activation)) {
-    cudnnActivationMode_t mode;
-    switch (m->activation) {
-      case AC_MODE_RELU:
-        mode = CUDNN_ACTIVATION_RELU;
-        break;
-      case AC_MODE_SIGMOID:
-        mode = CUDNN_ACTIVATION_SIGMOID;
-        break;
-      default:
-        // Unsupported activation mode
-        assert(false);
-    }
-    checkCUDNN(cudnnSetActivationDescriptor(
-        m->actiDesc, mode, CUDNN_PROPAGATE_NAN, 0.0));
-    checkCUDNN(cudnnSetTensor4dDescriptor(m->outputTensor,
-                                          CUDNN_TENSOR_NCHW,
-                                          ff_to_cudnn_datatype(m->output_type),
-                                          batch_size,
-                                          channel,
-                                          1,
-                                          1));
-  }
+  // todo: how to use allocator to allocate memory for float * one_ptr, how many
+  // bytes to allocate?
```

Comment:
Use the same allocation method that is in the original constructor for `LinearPerDeviceState`